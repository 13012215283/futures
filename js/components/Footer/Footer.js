import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { px2dp } from '../../utils/px2dp';

export default function Footer({ children }) {
  return (
    <View
      style={{
        position: 'absolute',
        bottom: px2dp(64),
        width: '100%',
      }}
    >
      <View
        style={{
          flexDirection: 'row',
          alignItems: 'center',
          justifyContent: 'center',
        }}
      >
        {React.Children.map(children, child => child)}
      </View>
    </View>
  );
}

Footer.propTypes = {
  children: PropTypes.node.isRequired,
};
