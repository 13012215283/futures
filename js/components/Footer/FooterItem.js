import React from 'react';
import PropTypes from 'prop-types';
import { View, StyleSheet, Text } from 'react-native';

import { colors, sizes } from '../themes';
import { px2dp } from '../../utils/px2dp';
import noop from '../../utils/noop';

export default function FooterItem({ text, onPress, type }) {
  return (
    <View>
      <Text
        style={type === 'separating' ? styles.separating : styles.footerText}
        onPress={onPress}
      >
        {text}
      </Text>
    </View>
  );
}

FooterItem.propTypes = {
  text: PropTypes.string.isRequired,
  type: PropTypes.string,
  onPress: PropTypes.func,
};

FooterItem.defaultProps = {
  type: '',
  onPress: noop,
};

const styles = StyleSheet.create({
  footerText: {
    color: colors[1004],
    fontSize: sizes.f1,
  },
  separating: {
    margin: px2dp(32),
  },
});
