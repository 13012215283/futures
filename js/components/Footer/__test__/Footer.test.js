import React from 'react';
import { shallow } from 'enzyme';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import Footer from '../Footer';

describe('<Footer/>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Footer>
          <Text>hello</Text>
        </Footer>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders multiple children correctly', () => {
    const tree = shallow(
      <Footer>
        <Text>hello</Text>
        <Text>world</Text>
      </Footer>
    );
    expect(tree.find(Text).length).toBe(2);
  });
});
