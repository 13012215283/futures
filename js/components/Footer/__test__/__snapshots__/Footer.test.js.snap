// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> renders correctly 1`] = `
<View
  style={
    Object {
      "bottom": 64,
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      hello
    </Text>
  </View>
</View>
`;
